src/components/LogButton.js

```diff
@@ -37,54 +40,60 @@ export class LogButton extends Component {
                 <Card.Content>
-                    <Title>My Habit</Title>
-                    <Paragraph>{this.props.count} / {this.props.limit}</Paragraph>
+                    <Title>{this.props.habit.title}</Title>
+                    <Paragraph>{this.props.habit.count} / {this.props.habit.limit}</Paragraph>
                     <Paragraph>Last logged: {this.props.updatedDisplay}</Paragraph>
                 </Card.Content>
                 <Card.Actions>
-                    <Button mode='contained' disabled={this.props.hasReachedLimit} onPress={this.props.incrementCount}>+</Button>
+                    <Button style={styles.plus} mode='contained' disabled={this.props.hasReachedLimit} onPress={this.props.incrementCount}>+</Button>
                     <Button mode='contained' onPress={this.showDialog}>Set Limit</Button>
                 </Card.Actions>
-                <Dialog visible={this.state.visible} onDismiss={this.hideDialog}>
-                    <Dialog.Title>Choose a daily limit</Dialog.Title>
-                    <Dialog.Content>
-                        <TextInput
-                            label="Daily limit"
-                            onChangeText={(limit) => this.setTempLimit(limit)}
-                        />
-                    </Dialog.Content>
-                    <Dialog.Actions>
-                        <Button onPress={this.hideDialog}>Cancel</Button>
-                        <Button onPress={this.setLimit}>Ok</Button>
-                    </Dialog.Actions>
-                </Dialog>
+                <Portal>
+                    <Dialog visible={this.state.visible} onDismiss={this.hideDialog}>
+                        <Dialog.Title>Choose a daily limit</Dialog.Title>
+                        <Dialog.Content>
+                            <TextInput
+                                label="Daily limit"
+                                onChangeText={(limit) => this.setTempLimit(limit)}
+                            />
+                        </Dialog.Content>
+                        <Dialog.Actions>
+                            <Button onPress={this.hideDialog}>Cancel</Button>
+                            <Button onPress={this.setLimit}>Ok</Button>
+                        </Dialog.Actions>
+                    </Dialog>
+                </Portal>
             </Card >
         );
     }
 };
 
-const mapStateToProps = (state) => ({
-    count: state.count.count,
-    updated: state.count.updated,
-    updatedDisplay: state.count.updated ? (new Date(state.count.updated)).toLocaleString() : 'Never',
-    hasReachedLimit: hasReachedDailyLimitSelector(state),
-    limit: state.count.limit
-})
+function mapStateToProps(state, ownProps) {
+    const habit = state.count.habits[ownProps.id];
+    const lastIncrementLog = habit ? habit.log.filter((log) => { return log.info.type == 'increment' }).slice(-1) : null;
+    var event = lastIncrementLog[0];
```

Comment:
I don't think we can get this case in the app with the way it's put together right now, but if `habit` was `null` then `lastIncrementLog` would also be `null` - so here we'd get `Uncaught TypeError: Cannot read property '0' of null`

If `lastIncrementLog` equalled `[]` when `habit` was `null` I think we'd be safe

Reply:
I hadn't thought of that - good catch. Have switched that now and also refactored this component to use hooks as well.